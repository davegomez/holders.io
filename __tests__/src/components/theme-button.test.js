import React from 'react'
import ThemeButton from '../../../src/components/theme-button'
import renderer from 'react-test-renderer'
import { indetity } from '../test-utils'

test('ThemeButton component active and no theme', () => {
  const component = renderer.create(
    <ThemeButton
      key='foo'
      active={ true }
      onClick={ indetity }
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('ThemeButton component active', () => {
  const component = renderer.create(
    <ThemeButton
      key='foo'
      theme='foo'
      active={ true }
      onClick={ indetity }
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('ThemeButton component inactive', () => {
  const component = renderer.create(
    <ThemeButton
      key='empty'
      theme='empty'
      active={ false }
      onClick={ indetity }
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
