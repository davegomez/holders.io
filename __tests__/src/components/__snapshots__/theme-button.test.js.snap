// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeButton component active 1`] = `
<button
  className="active"
  data-jsx={43889984}
  data-theme="foo"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('/static/img/themes/foo.jpg')",
    }
  }
/>
`;

exports[`ThemeButton component active 2`] = `
<button
  className="active"
  data-jsx={43889984}
  data-theme="foo"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('/static/img/themes/foo.jpg')",
    }
  }
/>
`;

exports[`ThemeButton component active and no theme 1`] = `
<button
  className="active"
  data-jsx={43889984}
  data-theme={null}
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('/static/img/themes/undefined.jpg')",
    }
  }
/>
`;

exports[`ThemeButton component active and no theme 2`] = `
<button
  className="active"
  data-jsx={43889984}
  data-theme={null}
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('/static/img/themes/undefined.jpg')",
    }
  }
/>
`;

exports[`ThemeButton component inactive 1`] = `
<button
  className={null}
  data-jsx={43889984}
  data-theme="empty"
  onClick={[Function]}
  style={null}
/>
`;

exports[`ThemeButton component inactive 2`] = `
<button
  className={null}
  data-jsx={43889984}
  data-theme="empty"
  onClick={[Function]}
  style={null}
/>
`;
